frontend/index.html

```diff
@@ -3,24 +3,51 @@
   <head>
     <meta charset="UTF-8" />
     <meta name="viewport" content="width=device-width, initial-scale=1.0" />
-    <!-- TAG INJECTION PLACEHOLDER -->
+    <title>{{ name }}</title>
+    <link rel="icon" href="{{ favicon_path }}" />
+    <meta name="description" content="{{ description }}">
+    <meta property="og:type" content="website">
+    <meta property="og:title" content="{{ name }}">
+    <meta property="og:description" content="{{ description }}">
+    <meta property="og:image" content="{{ og_image_url }}">
+    <meta property="og:url" content="{{ github }}">
+
+    <!-- Twitter Card data -->
+    <meta name="twitter:card" content="summary_large_image">
+    <meta name="twitter:title" content="{{ name }}">
+    <meta name="twitter:description" content="{{ description }}">
+    <meta name="twitter:image" content="{{ twitter_card_image_url }}">
+
+    <!-- Preconnect for performance -->
     <link rel="preconnect" href="https://fonts.googleapis.com" />
     <link rel="preconnect" href="https://fonts.gstatic.com" crossorigin />
-    <!-- FONT START -->
-    <link
-      href="https://fonts.googleapis.com/css2?family=Inter:wght@400;500;700&display=swap"
-      rel="stylesheet"
-    />
-    <!-- FONT END -->
-    <link
-      rel="stylesheet"
-      href="https://cdn.jsdelivr.net/npm/katex@0.16.8/dist/katex.min.css"
-    />
-    <!-- JS INJECTION PLACEHOLDER -->
-    <!-- CSS INJECTION PLACEHOLDER -->
-    <script>
-      const global = globalThis;
-    </script>
+
+    <!-- Conditionally include custom font or default Inter font -->
+    {% if custom_font %}
+      <link rel="stylesheet" href="{{ custom_font }}">
+    {% else %}
+      <link href="https://fonts.googleapis.com/css2?family=Inter:wght@400;500;700&display=swap" rel="stylesheet" />
+    {% endif %}
+
+    <!-- Include KaTeX CSS if needed -->
+    <link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/katex@0.16.8/dist/katex.min.css" />
+
+    <!-- Conditionally include custom CSS -->
+    {% if custom_css %}
+      <link rel="stylesheet" type="text/css" href="{{ custom_css }}">
+    {% endif %}
+
+    <!-- Theme Configuration Script -->
+    {% if theme %}
+      <script id="theme-data" type="application/json">
+        window.theme = {{ theme | tojson | safe }};
+      </script>
+    {% endif %}
+
+    <!-- Conditionally include custom JS -->
+    {% if custom_js %}
+      <script src="{{ custom_js }}" defer></script>
+    {% endif %}
   </head>
   <body>
     <div id="root"></div>
```

Comment:
It seems you have duplicated index.html to .j2 rather than renamed! 

If that is indeed the case, please, remove index.html in favour of the new version.

Reply:
No Mathijs, i had renamed the index.html -> index.html.j2. It was not duplicate, I left the "index.html.j2" file my mistake.  
Unfortunately, I'm not able to proceed with it. I just tried it again, and same error was reproduced.

<img width="905" alt="image" src="https://github.com/user-attachments/assets/9f5401f6-c0d4-43af-8862-6b416fcd3bee">

I checked all the reference, and I couldn't locate any frontend file that's directly referring this file template file.